diff --git a/src/Plugin/Map/OLMap/OLMap.php b/src/Plugin/Map/OLMap/OLMap.php
index 3c898d00..40ac6fd6 100644
--- a/src/Plugin/Map/OLMap/OLMap.php
+++ b/src/Plugin/Map/OLMap/OLMap.php
@@ -140,7 +140,7 @@ class OLMap extends Map {
     $i = 0;
     $data = array();
     $map_options = $this->getOptions();
-    /** @var \Drupal\openlayers\Types\Object $object */
+    /** @var \Drupal\openlayers\Types\Base $object */
     foreach ($this->getCollection()->getFlatList() as $object) {
       $weight = 0;
       if (isset($map_options['capabilities']['options']['table'][$object->getMachineName()])) {
diff --git a/src/Types/Collection.php b/src/Types/Collection.php
index 0960e146..a49f9bf0 100644
--- a/src/Types/Collection.php
+++ b/src/Types/Collection.php
@@ -9,7 +9,7 @@ namespace Drupal\openlayers\Types;
 use Drupal\Component\Plugin\PluginBase;
 use Drupal\openlayers\Component\Annotation\OpenlayersPlugin;
 use Drupal\openlayers\Openlayers;
-use Drupal\openlayers\Types\Object;
+use Drupal\openlayers\Types\Base;


 /**
@@ -26,7 +26,7 @@ class Collection extends PluginBase {
    *
    * @var BaseInterface[] $objects
    *   List of objects in this collection. The items have to be instances of
-   *   \Drupal\openlayers\Types\Object.
+   *   \Drupal\openlayers\Types\Base.
    */
   protected $objects = array();

diff --git a/src/Types/Component.php b/src/Types/Component.php
index c31c165a..a08d34c8 100644
--- a/src/Types/Component.php
+++ b/src/Types/Component.php
@@ -9,7 +9,7 @@ namespace Drupal\openlayers\Types;
 /**
  * Class Component.
  */
-abstract class Component extends Object implements ComponentInterface {
+abstract class Component extends Base implements ComponentInterface {
   /**
    * The array containing the options.
    *
diff --git a/src/Types/Control.php b/src/Types/Control.php
index 31fbd1be..b2d18722 100644
--- a/src/Types/Control.php
+++ b/src/Types/Control.php
@@ -9,7 +9,7 @@ namespace Drupal\openlayers\Types;
 /**
  * Class Control.
  */
-abstract class Control extends Object implements ControlInterface {
+abstract class Control extends Base implements ControlInterface {
   /**
    * The array containing the options.
    *
diff --git a/src/Types/Error.php b/src/Types/Error.php
index c4548001..560fd73b 100644
--- a/src/Types/Error.php
+++ b/src/Types/Error.php
@@ -8,7 +8,7 @@ namespace Drupal\openlayers\Types;

 use Drupal\Core\Logger\LoggerChannelInterface;
 use Drupal\openlayers\Component\Annotation\OpenlayersPlugin;
-use Drupal\openlayers\Types\Object;
+use Drupal\openlayers\Types\Base;
 use Drupal\service_container\Messenger\MessengerInterface;

 /**
@@ -25,7 +25,7 @@ use Drupal\service_container\Messenger\MessengerInterface;
  * Dummy class to avoid breaking the whole processing if a plugin class is
  * missing.
  */
-class Error extends Object implements ControlInterface, ComponentInterface, LayerInterface, SourceInterface, StyleInterface {
+class Error extends Base implements ControlInterface, ComponentInterface, LayerInterface, SourceInterface, StyleInterface {

   /**
    * @var string
diff --git a/src/Types/Interaction.php b/src/Types/Interaction.php
index 98fe4a55..8b99c930 100644
--- a/src/Types/Interaction.php
+++ b/src/Types/Interaction.php
@@ -9,7 +9,7 @@ namespace Drupal\openlayers\Types;
 /**
  * Class Interaction.
  */
-abstract class Interaction extends Object implements InteractionInterface {
+abstract class Interaction extends Base implements InteractionInterface {
   /**
    * The array containing the options.
    *
diff --git a/src/Types/Layer.php b/src/Types/Layer.php
index 65b39c7f..fa66582e 100644
--- a/src/Types/Layer.php
+++ b/src/Types/Layer.php
@@ -11,7 +11,7 @@ use Drupal\openlayers\Openlayers;
 /**
  * Class Layer.
  */
-abstract class Layer extends Object implements LayerInterface {
+abstract class Layer extends Base implements LayerInterface {
   /**
    * The array containing the options.
    *
diff --git a/src/Types/Map.php b/src/Types/Map.php
index 8e6d80fa..639caf29 100644
--- a/src/Types/Map.php
+++ b/src/Types/Map.php
@@ -11,7 +11,7 @@ use Drupal\openlayers\Openlayers;
 /**
  * Class Map.
  */
-abstract class Map extends Object implements MapInterface {
+abstract class Map extends Base implements MapInterface {
   /**
    * {@inheritdoc}
    *
diff --git a/src/Types/Object.php b/src/Types/Object.php
index e5d26635..f1f02b84 100644
--- a/src/Types/Object.php
+++ b/src/Types/Object.php
@@ -15,7 +15,7 @@ use Drupal\openlayers\Types\BaseInterface;
 /**
  * Class Object.
  */
-abstract class Object extends PluginBase implements BaseInterface {
+abstract class Base extends PluginBase implements BaseInterface {
   /**
    * A unique ID for the object.
    *
diff --git a/src/Types/Projection.php b/src/Types/Projection.php
index dc11714a..471c71b1 100644
--- a/src/Types/Projection.php
+++ b/src/Types/Projection.php
@@ -9,6 +9,6 @@ namespace Drupal\openlayers\Types;
 /**
  * Class Projection.
  */
-abstract class Projection extends Object implements ProjectionInterface {
+abstract class Projection extends Base implements ProjectionInterface {

 }
diff --git a/src/Types/Source.php b/src/Types/Source.php
index d98c51fa..727bfae3 100644
--- a/src/Types/Source.php
+++ b/src/Types/Source.php
@@ -9,7 +9,7 @@ namespace Drupal\openlayers\Types;
 /**
  * Class Source.
  */
-abstract class Source extends Object implements SourceInterface {
+abstract class Source extends Base implements SourceInterface {
   /**
    * The array containing the options.
    *
diff --git a/src/Types/Style.php b/src/Types/Style.php
index cf499154..d42c1edf 100644
--- a/src/Types/Style.php
+++ b/src/Types/Style.php
@@ -9,7 +9,7 @@ namespace Drupal\openlayers\Types;
 /**
  * Class Style.
  */
-abstract class Style extends Object implements StyleInterface {
+abstract class Style extends Base implements StyleInterface {
   /**
    * The array containing the options.
    *
